### PR TITLE
ERRAI-905: 4.0.0-SNAPSHOT: TranslationServiceGenerator creates flawed URLs for MessageBundleScanner

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/ERRAI-905

### CODE REVIEW

File: errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java [hunk 1]
```diff
@@ -261,7 +260,7 @@ private void addUrlsFromBundleAnnotations(final Collection<MetaClass> bundleAnno
         final URL classpathElement;
         final String pathRoot = getPathRoot(bundleClass, resource);
         try {
-          String urlString = new File(pathRoot).toURI().toURL().toString();
+          String urlString = new URI(pathRoot).toURL().toString();
```
Comment (manstis) [inline errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java]: Skip passing ```URL``` through ```File``` as ```file:/....``` is interpreted by ```toURI()``` as not absolute and hence the working folder path is appended to what ```File``` thinks is a relative path.

File: errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java [hunk 2]
```diff
@@ -289,7 +289,8 @@ private String getPathRoot(final MetaClass bundleClass, final URL resource) {
       // Do NOT use File.separatorChar here: Url.getPath() always uses forward-slashes
       relativePath = bundleClass.getPackageName().replace('.', '/');
 
-    return fullPath.substring(0, fullPath.indexOf(relativePath));
+    String pathRoot = fullPath.substring(0, fullPath.indexOf(relativePath));
```
Comment (manstis) [inline errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java]: I attempt to preserve the ```jar:file:/path/to/jar.jar!/``` semantics.. I we remove the JAR ```URL``` "wrapping" we'd also need to strip the trailing ```!``` e.g. ```jar:file:/path/to/jar.jar!/``` would become ```file:/path/to/jar.jar```.

**Comment by manstis** [comment]
I've commented the changes; the rest is code formatting...

File: errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java [hunk 3]
```diff
@@ -237,13 +236,13 @@ public String generate(TreeLogger logger, GeneratorContext context) {
     final Collection<URL> urls = new HashSet<URL>();
 
     addUrlsFromBundleAnnotations(bundleAnnotatedClasses, urls);
-    addUrlsFromErraiAppProperties(urls);
+    addUrlsFromErraiAppProperties( urls );
```
Comment (mbarkley) [inline errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java]: Minor formatting issue: There's no space around method arguments in the Errai code style.

File: errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java [hunk 4]
```diff
@@ -237,13 +236,13 @@ public String generate(TreeLogger logger, GeneratorContext context) {
...
     addUrlsFromBundleAnnotations(bundleAnnotatedClasses, urls);
-    addUrlsFromErraiAppProperties(urls);
+    addUrlsFromErraiAppProperties( urls );
 
     return urls;
   }
 
   private void addUrlsFromErraiAppProperties(final Collection<URL> urls) {
-    urls.addAll(MetaDataScanner.getConfigUrls());
+    urls.addAll( MetaDataScanner.getConfigUrls() );
```
Comment (mbarkley) [inline errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java]: Minor formatting issue: There's no space around method arguments in the Errai code style.

**Comment by manstis** [comment]
@mbarkley Darn it.. I reverted my commit; and *only* edited the handful of lines required yet Git still insists on showing a ton of formatting differences...... let me try ```vi`` instead of InteliJ (oh, BTW, my line endings are set as unix-style in InteliJ).

**Comment by manstis** [comment]
@mbarkley heh. now this is taking the mickey! Edited with a Linux TextEditor (OK, not ```vi```) and it's smurfed all the line-endings for me... Can I just paste the lines changed and let you merge?!?!

**Comment by mbarkley** [comment]
@manstis Don't worry about it. I will see if I can clean it up on my machine before cherry-picking.

**Comment by mbarkley** [comment]
master: 9fc9f820b88611a5c71a663d479b224bb67f4e8d

**Pull request closed without merging**